pyam_analysis/core.py

```diff
@@ -765,21 +787,15 @@ def read_ix(ix, regions=None):
     regions: list
         list of regions to be loaded from the database snapshot
     """
-    if not has_ix:
-        error = 'this option depends on the ixmp package'
-        raise SystemError(error)
     if isinstance(ix, ixmp.TimeSeries):
```

Comment:
normally, its best to put the error checking first (`if condition: raise Error()`) and then have the rest of the function and top-level indentation.

Reply:
I took that check out because `read_ix()` will only be called internally from functions that check for an `ixmp.TimeSeries` instance - and if someone calls this function from outside, they better know which packages they have installed and imported...

Reply:
Yes, I'm referring to the below else block: 
```
     else:
         error = 'arg ' + ix + ' not recognized as valid ixmp class'
         raise ValueError(error)
```

Which can be moved to the top of the function, and all other indentation at top-level.